oak/server/rust/oak_runtime/src/runtime/mod.rs

```diff
@@ -209,8 +326,11 @@ impl Runtime {
     /// available.
     pub fn wait_on_channels(
         &self,
+        node: NodeId,
         readers: &[Option<Handle>],
     ) -> Result<Vec<ChannelReadStatus>, OakStatus> {
+        let readers = self.filter_optional_handles(node, readers.iter());
```

Comment:
Same comment here about this being too lenient.

Reply:
I've changed filter_optional_handles to validate_handles_access, which returns `ErrBadHandle` on any handle the node doesnt have access to (and errors immediately)